### PR TITLE
[Authentication] Add Feature Component's login page and signup page

### PR DESCRIPTION
**Pull request by RenaufalAlrafid**

### CODE REVIEW

**Comment by gilangpersada** [comment]
- Description PR nya mana?
- Title PR depannya pake `[Authentication] ....`
- Screenshot nya mana?
- Assignee nya kosong

File: lib/Scenes/Page/LoginPage/componnent/google_button.dart [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import 'package:flutter/material.dart';
+import 'package:timeboxing/Shared/Extension/colors_style_extension.dart';
+import 'package:timeboxing/Shared/Extension/text_style_extension.dart';
```
Comment (gilangpersada) [inline lib/Scenes/Page/LoginPage/componnent/google_button.dart]: pake `package:timeboxing/Shared/Extension/extension_barrel.dart`

File: lib/Scenes/Page/LoginPage/componnent/intro_text.dart [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/material.dart';
+import 'package:timeboxing/Shared/Extension/colors_style_extension.dart';
+import 'package:timeboxing/Shared/Extension/text_style_extension.dart';
```
Comment (gilangpersada) [inline lib/Scenes/Page/LoginPage/componnent/intro_text.dart]: pake package:timeboxing/Shared/Extension/extension_barrel.dart

File: lib/Scenes/Page/LoginPage/componnent/login_form.dart [hunk 3]
```diff
@@ -0,0 +1,157 @@
+import 'package:flutter/material.dart';
+import 'package:timeboxing/Shared/Extension/colors_style_extension.dart';
+import 'package:timeboxing/Shared/Extension/text_style_extension.dart';
```
Comment (gilangpersada) [inline lib/Scenes/Page/LoginPage/componnent/login_form.dart]: pake package:timeboxing/Shared/Extension/extension_barrel.dart

**Comment by RenaufalAlrafid** [comment]
Component for login and sign up page without model

**Comment by RenaufalAlrafid** [comment]
Screenshot 2023-07-30 192630

**Comment by RenaufalAlrafid** [comment]
Screenshot 2023-07-30 192658

File: macos/Flutter/GeneratedPluginRegistrant.swift [hunk 4]
```diff
@@ -5,6 +5,8 @@
 import FlutterMacOS
 import Foundation
 
+import firebase_core
 
 func RegisterGeneratedPlugins(registry: FlutterPluginRegistry) {
+  FLTFirebaseCorePlugin.register(with: registry.registrar(forPlugin: "FLTFirebaseCorePlugin"))
 }
```
Comment (gilangpersada) [inline macos/Flutter/GeneratedPluginRegistrant.swift]: ini file apa hayo

File: windows/flutter/generated_plugin_registrant.cc [hunk 5]
```diff
@@ -6,6 +6,9 @@
 
 #include "generated_plugin_registrant.h"
 
+#include <firebase_core/firebase_core_plugin_c_api.h>
 
 void RegisterPlugins(flutter::PluginRegistry* registry) {
+  FirebaseCorePluginCApiRegisterWithRegistrar(
+      registry->GetRegistrarForPlugin("FirebaseCorePluginCApi"));
 }
```
Comment (gilangpersada) [inline windows/flutter/generated_plugin_registrant.cc]: ini file apa hayo

File: windows/flutter/generated_plugins.cmake [hunk 6]
```diff
@@ -3,6 +3,7 @@
 #
 
 list(APPEND FLUTTER_PLUGIN_LIST
+  firebase_core
 )
 
 list(APPEND FLUTTER_FFI_PLUGIN_LIST
```
Comment (gilangpersada) [inline windows/flutter/generated_plugins.cmake]: ini file apa hayo

**Comment by RenaufalAlrafid** [comment]
image

File: lib/Scenes/Page/LoginPage/componnent/google_button.dart (hunk 1 above)
Comment (RenaufalAlrafid) [reply lib/Scenes/Page/LoginPage/componnent/google_button.dart]: resolve

File: lib/Scenes/Page/LoginPage/componnent/intro_text.dart (hunk 2 above)
Comment (RenaufalAlrafid) [reply lib/Scenes/Page/LoginPage/componnent/intro_text.dart]: resolve

File: lib/Scenes/Page/LoginPage/componnent/login_form.dart (hunk 3 above)
Comment (RenaufalAlrafid) [reply lib/Scenes/Page/LoginPage/componnent/login_form.dart]: resolve

File: macos/Flutter/GeneratedPluginRegistrant.swift (hunk 4 above)
Comment (RenaufalAlrafid) [reply macos/Flutter/GeneratedPluginRegistrant.swift]: resolve

File: windows/flutter/generated_plugin_registrant.cc (hunk 5 above)
Comment (RenaufalAlrafid) [reply windows/flutter/generated_plugin_registrant.cc]: resolve

File: windows/flutter/generated_plugins.cmake (hunk 6 above)
Comment (RenaufalAlrafid) [reply windows/flutter/generated_plugins.cmake]: resolve

**Review by gilangpersada — APPROVED** [review]

**Review by irfandigalih — APPROVED** [review]
GG bang pal

**Pull request merged**